# Importation des bibliothèques

In [13]:
import pandas as pd

# Constantes

In [14]:
# chemin pour accéder aux données
FILE_PATH = '../src/data/raw/donnees_immobilieres.parquet'

# Application

| Nom de colonne                 | Description |
|--------------------------------|-------------|
| region                         | Région où se situe le bien |
| ue                             | Identifiant de l’Unité économique (Site). À compter du 31/10/2015, remplacé par l’ID unique du bien |
| id                             | Identifiant unique du bien |
| type                           | Usage majoritaire du bien |
| fonction                       | Usage majoritaire détaillé du bien |
| designation_batiment_terrain   | Désignation du bâtiment ou terrain |
| dept                           | Code du département de situation du bien |
| adresse                        | Adresse normalisée du bien |
| code_postal                    | Code postal de situation du bien |
| ville                          | Ville de situation du bien |
| pays                           | Pays de situation du bien |
| ministere                      | Ministère occupant le bien |
| date_inventaire                | Date de l’inventaire |

In [15]:
# Chargement du DataFrame depuis le fichier Parquet 
df = pd.read_parquet(FILE_PATH)

In [16]:
df.head()

,region,ue,id,type,fonction,designation_batiment_terrain,dept,adresse,code_postal,ville,pays,ministere,date_inventaire
0,POITOU,178661,IG00100077264,Espace naturel,Terrain en friche,XYNTHIA17091POIRIER GENEVIEVE,17,21 R DU 14 JUILLET,17230,Charron,France,Ecologie - Développement durable - Energie - A...,2017-12
1,NORD-PAS DE CALAIS,132144,IG00100007106,Espace naturel,Terrain en friche,QUAI AU BOIS - Dunkerque,59,QU AU BOIS,59140,Dunkerque,France,Ecologie - Développement durable - Energie - A...,2017-03
2,ILE DE France,143844,IB00100024371,Bureau,Immeuble de bureaux,BATIMENT PRINCIPAL - QUAI D'ORSAY - MAE,75,37 Quai D ORSAY,75000,PARIS 07,France,Affaires étrangères et européennes.,2013-05
3,NORD-PAS DE CALAIS,145878,IB00100013660,Bureau,Hôtel des impôts,SIP-SIE,59,195 Rue DE ROUBAIX,59500,DOUAI,France,"Budget, comptes publics et fonction publique.",2013-05
4,PACA,135078,IG00100016428,Espace naturel,Terrain en friche,ANCIENNE BASE AERO NAVALE,13,Lieu-dit LA BASE ET LE MOULIN DE GO,13130,BERRE L ETANG,France,"Ecologie, développement et aménagement durables.",2013-05


In [17]:
# Dimensions du DataFrame 
print(f"Nombre de lignes : {df.shape[0]}")
print(f"Nombre de colonnes : {df.shape[1]}")

Nombre de lignes : 1000
Nombre de colonnes : 13


In [18]:
# Informations générales sur le DataFrame 
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 1000 entries, 0 to 999
Data columns (total 13 columns):
 #   Column                        Non-Null Count  Dtype
---  ------                        --------------  -----
 0   region                        1000 non-null   str  
 1   ue                            970 non-null    str  
 2   id                            1000 non-null   str  
 3   type                          1000 non-null   str  
 4   fonction                      1000 non-null   str  
 5   designation_batiment_terrain  1000 non-null   str  
 6   dept                          1000 non-null   str  
 7   adresse                       693 non-null    str  
 8   code_postal                   720 non-null    str  
 9   ville                         720 non-null    str  
 10  pays                          1000 non-null   str  
 11  ministere                     1000 non-null   str  
 12  date_inventaire               1000 non-null   str  
dtypes: str(13)
memory usage: 296.5 KB


In [19]:
# Statistiques descriptives
df.describe()

,region,ue,id,type,fonction,designation_batiment_terrain,dept,adresse,code_postal,ville,pays,ministere,date_inventaire
count,1000,970,1000,1000,1000,1000,1000,693,720,720,1000,1000,1000
unique,25,835,994,17,67,685,107,678,527,613,16,38,6
top,ILE DE France,110056,IG00100017956,Espace naturel,biens du ministère de l'Intérieur,biens du ministère de l'Intérieur,13,LD LA PLAGE,99000,Paris 7e Arrondissement,France,Ecologie - Développement durable - Energie - A...,2013-05
freq,118,8,2,408,277,277,45,3,13,7,982,269,189


# 🏢 Interprétation des statistiques descriptives (df.describe)

## 1. Vue d’ensemble
Le dataset contient **10 000 biens immobiliers publics** avec 13 colonnes :
- Identifiants (`ue`, `id`)
- Localisation (`region`, `dept`, `adresse`, `code_postal`, `ville`, `pays`)
- Usage (`type`, `fonction`, `designation_batiment_terrain`)
- Ministère occupant (`ministere`)
- Date de l’inventaire (`date_inventaire`)

---

## 2. Analyse par colonnes

### Identifiants
- **id** : 10 000 valeurs mais seulement 9 041 uniques → certains identifiants apparaissent plusieurs fois (doublons à traiter).
- **ue** : 9 703 valeurs renseignées, 5 110 uniques. Valeurs manquantes (≈3 %) car remplacé par l’`id` après 2015.

### Localisation
- **region** : 25 valeurs distinctes, cohérent avec les anciennes régions françaises. Valeur la plus fréquente : *Île-de-France* (1171 biens).
- **dept** : 110 départements distincts. 2 valeurs manquantes.
- **ville** : 6 972 valeurs renseignées (≈70 %). 3 775 villes distinctes. Valeur la plus fréquente : *La Faute-sur-Mer* (54 occurrences).
- **code_postal** : 6 985 valeurs renseignées (≈70 %), 2 310 distincts.
- **adresse** : seulement 6 712 adresses renseignées (≈67 %), 5 697 distinctes.
- **pays** : quasi exclusivement *France* (98,2 %).

### Usage
- **type** : 18 catégories différentes. Valeur la plus fréquente : *Espace naturel* (3 887 biens).
- **fonction** : 107 usages détaillés. Valeur la plus fréquente : *biens du ministère de l’Intérieur* (2 989 occurrences).
- **designation_batiment_terrain** : 5 515 désignations distinctes, mais redondance forte avec *biens du ministère de l’Intérieur* (2 989 occurrences).

### Ministère
- **ministere** : 46 ministères distincts. Le plus représenté : *Écologie - Développement durable - Énergie ...* (2 518 biens).

### Date d’inventaire
- **date_inventaire** : seulement 6 valeurs distinctes (ex. 2013-05, 2014-03, 2015-10…). Cela correspond à des campagnes d’inventaire ponctuelles.

---

## 3. Points clés pour le nettoyage
- 🔍 **Doublons** : vérifier `id`, qui devrait être unique.
- ⚠️ **Valeurs manquantes** : fortes proportions sur `adresse`, `code_postal`, `ville`.
- 🧹 **Normalisation** :
  - Harmoniser la casse (`ville`, `region`, `ministere`).
  - Nettoyer les désignations redondantes.
- 📊 **Colonnes peu informatives** :
  - `pays` (quasi toujours *France*).
  - `date_inventaire` (peu de valeurs distinctes).

In [ ]:
# Affichage des valeurs distinctes pour les colonnes d'intérêt
for col_name in ["type", "fonction", "region", "dept", "ministere"]:
    valeurs_uniques = df[col_name].dropna().unique()
    valeurs_triees = sorted(valeurs_uniques)
    print(f"\n{'='*50}")
    print(f"  {col_name.upper()} ({len(valeurs_triees)} valeurs)")
    print(valeurs_triees)


  TYPE (17 valeurs)
['Bureau', 'Bât. enseignement ou sport', "Bâtiment agricole ou d'élevage", 'Bâtiment culturel', "Bâtiment d'enseignement ou de sport", 'Bâtiment sanitaire ou social', 'Bâtiment technique', 'Commerce', 'Edifice du culte', 'Espace aménagé', 'Espace naturel', 'Monument et mémorial', "Ouvrage d'art des réseaux et voiries", "Ouvrages d'art réseaux voiries", 'Réseaux et voirie', 'Réseaux et voiries', "biens du ministère de l'Intérieur"]

  FONCTION (67 valeurs)
['Aire de stationnement', 'Ambassade', 'Atelier réparation, entretien', 'Bibliothèque', 'Blockhaus', "Bât. d'enseignement supérieur", "Bât. d'instruction/formation", 'Bât. restauration collective', 'Bât. social avec hébergement', 'Bât. social sans hébergement', "Bâtiment d'élevage", 'Bâtiment de stockage', 'Bâtiment pénitentiaire', 'Bâtiment scientifique', 'Bâtiment scolaire/enseignement', 'Bâtiment technique', 'Cathédrale', 'Centre Informatique', "Centre d'information et d'orientation", 'Centre de loisirs', 'Cent

In [21]:
# exploration valeur unique "region"
df["region"].value_counts(normalize=True) * 100

region
ILE DE France           11.8
PACA                     9.6
D.O.M                    7.9
BRETAGNE                 7.4
NORD-PAS DE CALAIS       6.8
MIDI-PYRENNEES           5.5
RHONE ALPES              5.2
POITOU                   5.0
PAYS DE LOIRE            5.0
AQUITAINE                4.6
LANGUEDOC-ROUSSILLON     3.6
LORRAINE                 3.1
PICARDIE                 3.1
CENTRE                   2.9
AUVERGNE                 2.5
HAUTE NORMANDIE          2.1
BOURGOGNE                2.1
ETRANGER                 1.8
CHAMPAGNE-ARDENNES       1.8
BASSE NORMANDIE          1.8
FRANCHE COMTE            1.8
ALSACE                   1.8
LIMOUSIN                 1.4
CORSE                    1.1
C.O.M                    0.3
Name: proportion, dtype: float64

In [22]:
# exploration valeur unique "type"
df["type"].value_counts(normalize=True) * 100

type
Espace naturel                          40.8
biens du ministère de l'Intérieur       27.7
Bureau                                  11.9
Bâtiment technique                       7.6
Bâtiment sanitaire ou social             2.3
Ouvrage d'art des réseaux et voiries     2.2
Espace aménagé                           1.8
Bât. enseignement ou sport               1.6
Bâtiment culturel                        0.7
Bâtiment agricole ou d'élevage           0.6
Réseaux et voirie                        0.6
Ouvrages d'art réseaux voiries           0.5
Edifice du culte                         0.4
Réseaux et voiries                       0.4
Bâtiment d'enseignement ou de sport      0.4
Monument et mémorial                     0.3
Commerce                                 0.2
Name: proportion, dtype: float64

In [23]:
# exploration valeur unique "fonction"
df["fonction"].value_counts(normalize=True) * 100

fonction
biens du ministère de l'Intérieur    27.7
Terrain en friche                    22.6
Terrain délaissé                     12.4
Immeuble de bureaux                   8.1
Plage                                 2.1
                                     ... 
Maison cantonnière                    0.1
Bâtiment scientifique                 0.1
Foyer d'hébergement                   0.1
Terrain de jeux ou de loisirs         0.1
Atelier réparation, entretien         0.1
Name: proportion, Length: 67, dtype: float64

In [24]:
# exploration valeur unique "ville"
df["ville"].value_counts(normalize=True) * 100

ville
Paris 7e Arrondissement     0.972222
Toulouse                    0.972222
Charron                     0.833333
La Faute-sur-Mer            0.694444
L'Aiguillon-sur-Mer         0.555556
                              ...   
Paris 16e Arrondissement    0.138889
Barberaz                    0.138889
AYGUEMORTE LES GRAVES       0.138889
Arnas                       0.138889
Wasquehal                   0.138889
Name: proportion, Length: 613, dtype: float64

In [25]:
# exploration valeur unique "pays"
df["pays"].value_counts(normalize=True) * 100

pays
France                       98.2
Algérie                       0.2
Laos                          0.2
Liban                         0.2
République du Liban           0.1
Roumanie                      0.1
Syrie                         0.1
Ethiopie                      0.1
Djibouti                      0.1
Uruguay                       0.1
Israël                        0.1
République arabe syrienne     0.1
Bolivie                       0.1
Guinée                        0.1
Belgique                      0.1
Chili                         0.1
Name: proportion, dtype: float64

In [26]:
# exploration valeur unique "ministere"
df["ministere"].value_counts(normalize=True) * 100

ministere
Ecologie - Développement durable - Energie - Aménagement du territoire    26.9
Intérieur - Outre-mer - Collectivités territoriales                       18.1
France Domaine                                                            10.1
Ecologie, développement et aménagement durables.                           7.9
Ecologie, Energie, Dév.dur., Amt.ter.                                      7.6
Intérieur, outre-mer et collectivités territoriales.                       5.4
Intérieur, Outre-mer et Collec. terr.                                      4.2
Justice                                                                    4.1
Budget - Comptes publics - Fonction publique                               2.5
Culture                                                                    1.5
Budget, comptes publics et fonction publique.                              1.3
Education nationale - Enseignement supérieur - Recherche                   1.0
Multi-occupants                           

In [27]:
# exploration valeur unique "date_inventaire"
df["date_inventaire"].value_counts(normalize=True) * 100

date_inventaire
2013-05    18.9
2014-03    18.2
2017-12    16.3
2017-03    16.2
2015-10    15.5
2018-12    14.9
Name: proportion, dtype: float64

In [28]:
# exploration des valeurs manquantes dans la colonne 'code_postal'
df[df['code_postal'].isnull()]

,region,ue,id,type,fonction,designation_batiment_terrain,dept,adresse,code_postal,ville,pays,ministere,date_inventaire
10,LANGUEDOC-ROUSSILLON,e9a9abcb36f3e5e6686e150ed6bb7214,00ab869cabec0c3a83ea9a2a68a59643,biens du ministère de l'Intérieur,biens du ministère de l'Intérieur,biens du ministère de l'Intérieur,34,NaN,NaN,NaN,France,"Intérieur, Outre-mer et Collec. terr.",2014-03
11,CENTRE,e5b94bde60dfec432b28c4292b657c07,90d3c753538128e4285962d69317324c,biens du ministère de l'Intérieur,biens du ministère de l'Intérieur,biens du ministère de l'Intérieur,36,NaN,NaN,NaN,France,"Intérieur, Outre-mer et Collec. terr.",2014-03
12,D.O.M,e5be0b61a8e774825ddec79a508849d6,687b167e0d0e269b8d1c6367e51a55b1,biens du ministère de l'Intérieur,biens du ministère de l'Intérieur,biens du ministère de l'Intérieur,973,NaN,NaN,NaN,France,Intérieur - Outre-mer - Collectivités territor...,2017-03
24,PAYS DE LOIRE,b600f17d0a6a429e2076bd65cb0d8d87,09c2fd1c19396373d3af79d0f32770a1,biens du ministère de l'Intérieur,biens du ministère de l'Intérieur,biens du ministère de l'Intérieur,44,NaN,NaN,NaN,France,"Intérieur, Outre-mer et Collec. terr.",2014-03
25,LANGUEDOC-ROUSSILLON,5dd5d10d1539b9ab482ce04cbc1322e9,d5f3dad4ece5a8e45bca269c4b7a7858,biens du ministère de l'Intérieur,biens du ministère de l'Intérieur,biens du ministère de l'Intérieur,11,NaN,NaN,NaN,France,"Intérieur, Outre-mer et Collec. terr.",2014-03
...,...,...,...,...,...,...,...,...,...,...,...,...,...
962,D.O.M,2553fbda3251faeb8f8fcf83fc21d1a9,ff40869c087b9d8506166b37eb5609b0,biens du ministère de l'Intérieur,biens du ministère de l'Intérieur,biens du ministère de l'Intérieur,988,NaN,NaN,NaN,France,Intérieur - Outre-mer - Collectivités territor...,2017-12
963,D.O.M,1e39e97c3fd6edf21dd1fbe2d4b9ed16,be167122b44eb96189bd0c87d57c393b,biens du ministère de l'Intérieur,biens du ministère de l'Intérieur,biens du ministère de l'Intérieur,973,NaN,NaN,NaN,France,Intérieur - Outre-mer - Collectivités territor...,2018-12
964,POITOU,b827679599599f60911c6176bd5917cb,6a88c979a109bd8861284bc4f239d5c9,biens du ministère de l'Intérieur,biens du ministère de l'Intérieur,biens du ministère de l'Intérieur,17,NaN,NaN,NaN,France,Intérieur - Outre-mer - Collectivités territor...,2018-12
965,RHONE ALPES,44d9de4180f8819fc03bfab9e224a7f1,6271cf0c700ebd359615ee9256bc868e,biens du ministère de l'Intérieur,biens du ministère de l'Intérieur,biens du ministère de l'Intérieur,69,NaN,NaN,NaN,France,"Intérieur, Outre-mer et Collec. terr.",2014-03


In [29]:
# pourcentage de valeurs manquantes dans la colonne 'code_postal' par "ministere"
df[df['code_postal'].isnull()]['ministere'].value_counts(normalize=True) * 100

ministere
Intérieur - Outre-mer - Collectivités territoriales                       64.642857
Intérieur, outre-mer et collectivités territoriales.                      19.285714
Intérieur, Outre-mer et Collec. terr.                                     15.000000
France Domaine                                                             0.714286
Ecologie - Développement durable - Energie - Aménagement du territoire     0.357143
Name: proportion, dtype: float64

In [30]:
# exploration des valeurs manquantes dans la colonne 'adresse'
df[df['adresse'].isnull()]

,region,ue,id,type,fonction,designation_batiment_terrain,dept,adresse,code_postal,ville,pays,ministere,date_inventaire
10,LANGUEDOC-ROUSSILLON,e9a9abcb36f3e5e6686e150ed6bb7214,00ab869cabec0c3a83ea9a2a68a59643,biens du ministère de l'Intérieur,biens du ministère de l'Intérieur,biens du ministère de l'Intérieur,34,NaN,NaN,NaN,France,"Intérieur, Outre-mer et Collec. terr.",2014-03
11,CENTRE,e5b94bde60dfec432b28c4292b657c07,90d3c753538128e4285962d69317324c,biens du ministère de l'Intérieur,biens du ministère de l'Intérieur,biens du ministère de l'Intérieur,36,NaN,NaN,NaN,France,"Intérieur, Outre-mer et Collec. terr.",2014-03
12,D.O.M,e5be0b61a8e774825ddec79a508849d6,687b167e0d0e269b8d1c6367e51a55b1,biens du ministère de l'Intérieur,biens du ministère de l'Intérieur,biens du ministère de l'Intérieur,973,NaN,NaN,NaN,France,Intérieur - Outre-mer - Collectivités territor...,2017-03
24,PAYS DE LOIRE,b600f17d0a6a429e2076bd65cb0d8d87,09c2fd1c19396373d3af79d0f32770a1,biens du ministère de l'Intérieur,biens du ministère de l'Intérieur,biens du ministère de l'Intérieur,44,NaN,NaN,NaN,France,"Intérieur, Outre-mer et Collec. terr.",2014-03
25,LANGUEDOC-ROUSSILLON,5dd5d10d1539b9ab482ce04cbc1322e9,d5f3dad4ece5a8e45bca269c4b7a7858,biens du ministère de l'Intérieur,biens du ministère de l'Intérieur,biens du ministère de l'Intérieur,11,NaN,NaN,NaN,France,"Intérieur, Outre-mer et Collec. terr.",2014-03
...,...,...,...,...,...,...,...,...,...,...,...,...,...
963,D.O.M,1e39e97c3fd6edf21dd1fbe2d4b9ed16,be167122b44eb96189bd0c87d57c393b,biens du ministère de l'Intérieur,biens du ministère de l'Intérieur,biens du ministère de l'Intérieur,973,NaN,NaN,NaN,France,Intérieur - Outre-mer - Collectivités territor...,2018-12
964,POITOU,b827679599599f60911c6176bd5917cb,6a88c979a109bd8861284bc4f239d5c9,biens du ministère de l'Intérieur,biens du ministère de l'Intérieur,biens du ministère de l'Intérieur,17,NaN,NaN,NaN,France,Intérieur - Outre-mer - Collectivités territor...,2018-12
965,RHONE ALPES,44d9de4180f8819fc03bfab9e224a7f1,6271cf0c700ebd359615ee9256bc868e,biens du ministère de l'Intérieur,biens du ministère de l'Intérieur,biens du ministère de l'Intérieur,69,NaN,NaN,NaN,France,"Intérieur, Outre-mer et Collec. terr.",2014-03
966,ILE DE France,1cc1e8bdaa24690eb712d0265a6122dc,9ed11e915d6931f33a4b889c093a4644,biens du ministère de l'Intérieur,biens du ministère de l'Intérieur,biens du ministère de l'Intérieur,93,NaN,NaN,NaN,France,Intérieur - Outre-mer - Collectivités territor...,2015-10


In [31]:
# pourcentage de valeurs manquantes dans la colonne 'adresse' par "ministere"
df[df['adresse'].isnull()]['ministere'].value_counts(normalize=True) * 100

ministere
Intérieur - Outre-mer - Collectivités territoriales                       58.957655
Intérieur, outre-mer et collectivités territoriales.                      17.589577
Intérieur, Outre-mer et Collec. terr.                                     13.680782
Ecologie - Développement durable - Energie - Aménagement du territoire     5.211726
France Domaine                                                             1.954397
Ecologie, Energie, Dév.dur., Amt.ter.                                      0.977199
Affaires étrangères et européennes                                         0.325733
Agriculture                                                                0.325733
Ecologie, développement et aménagement durables.                           0.325733
Agriculture et pêche.                                                      0.325733
Culture                                                                    0.325733
Name: proportion, dtype: float64

In [32]:
# exploration des valeurs manquantes dans la colonne 'ville'
df[df['ville'].isnull()]

,region,ue,id,type,fonction,designation_batiment_terrain,dept,adresse,code_postal,ville,pays,ministere,date_inventaire
10,LANGUEDOC-ROUSSILLON,e9a9abcb36f3e5e6686e150ed6bb7214,00ab869cabec0c3a83ea9a2a68a59643,biens du ministère de l'Intérieur,biens du ministère de l'Intérieur,biens du ministère de l'Intérieur,34,NaN,NaN,NaN,France,"Intérieur, Outre-mer et Collec. terr.",2014-03
11,CENTRE,e5b94bde60dfec432b28c4292b657c07,90d3c753538128e4285962d69317324c,biens du ministère de l'Intérieur,biens du ministère de l'Intérieur,biens du ministère de l'Intérieur,36,NaN,NaN,NaN,France,"Intérieur, Outre-mer et Collec. terr.",2014-03
12,D.O.M,e5be0b61a8e774825ddec79a508849d6,687b167e0d0e269b8d1c6367e51a55b1,biens du ministère de l'Intérieur,biens du ministère de l'Intérieur,biens du ministère de l'Intérieur,973,NaN,NaN,NaN,France,Intérieur - Outre-mer - Collectivités territor...,2017-03
24,PAYS DE LOIRE,b600f17d0a6a429e2076bd65cb0d8d87,09c2fd1c19396373d3af79d0f32770a1,biens du ministère de l'Intérieur,biens du ministère de l'Intérieur,biens du ministère de l'Intérieur,44,NaN,NaN,NaN,France,"Intérieur, Outre-mer et Collec. terr.",2014-03
25,LANGUEDOC-ROUSSILLON,5dd5d10d1539b9ab482ce04cbc1322e9,d5f3dad4ece5a8e45bca269c4b7a7858,biens du ministère de l'Intérieur,biens du ministère de l'Intérieur,biens du ministère de l'Intérieur,11,NaN,NaN,NaN,France,"Intérieur, Outre-mer et Collec. terr.",2014-03
...,...,...,...,...,...,...,...,...,...,...,...,...,...
962,D.O.M,2553fbda3251faeb8f8fcf83fc21d1a9,ff40869c087b9d8506166b37eb5609b0,biens du ministère de l'Intérieur,biens du ministère de l'Intérieur,biens du ministère de l'Intérieur,988,NaN,NaN,NaN,France,Intérieur - Outre-mer - Collectivités territor...,2017-12
963,D.O.M,1e39e97c3fd6edf21dd1fbe2d4b9ed16,be167122b44eb96189bd0c87d57c393b,biens du ministère de l'Intérieur,biens du ministère de l'Intérieur,biens du ministère de l'Intérieur,973,NaN,NaN,NaN,France,Intérieur - Outre-mer - Collectivités territor...,2018-12
964,POITOU,b827679599599f60911c6176bd5917cb,6a88c979a109bd8861284bc4f239d5c9,biens du ministère de l'Intérieur,biens du ministère de l'Intérieur,biens du ministère de l'Intérieur,17,NaN,NaN,NaN,France,Intérieur - Outre-mer - Collectivités territor...,2018-12
965,RHONE ALPES,44d9de4180f8819fc03bfab9e224a7f1,6271cf0c700ebd359615ee9256bc868e,biens du ministère de l'Intérieur,biens du ministère de l'Intérieur,biens du ministère de l'Intérieur,69,NaN,NaN,NaN,France,"Intérieur, Outre-mer et Collec. terr.",2014-03


In [33]:
# pourcentage de valeurs manquantes dans la colonne 'ville' par "ministere"
df[df['ville'].isnull()]['ministere'].value_counts(normalize=True) * 100

ministere
Intérieur - Outre-mer - Collectivités territoriales                       64.642857
Intérieur, outre-mer et collectivités territoriales.                      19.285714
Intérieur, Outre-mer et Collec. terr.                                     15.000000
France Domaine                                                             0.714286
Ecologie - Développement durable - Energie - Aménagement du territoire     0.357143
Name: proportion, dtype: float64